File: examples/example-58/ruby/features/step_definitions/mobile.rb
require 'rspec'

Given('the app is opened') do
    @appium.start_driver
    @mydemoapp = MyDemoApp.new(@appium)
end

When('the {string} is clicked') do |item|
    wait = Selenium::WebDriver::Wait.new(timeout: 10)
    element = wait.until { @mydemoapp.find_element(item) }
    element.click
end

Then('the {string} should be displayed') do |item|
    wait = Selenium::WebDriver::Wait.new(timeout: 10)
    wait.until do
        found_element = @mydemoapp.find_element(item)
        found_element if found_element.displayed?
    end
end

File: examples/example-58/js/features/step_definitions/mobile.js
const { Given, When, Then } = require('@cucumber/cucumber');
const assert = require('assert');

Given('the app is opened', async function () { /* noop */ });

When('the {string} is clicked', async function (item) {
    const element = await this.appium.$(this.mydemoapp[item]);
    await element.click();
});

Then('the {string} should be displayed', async function (item) {
    const element = await this.appium.$(this.mydemoapp[item]);
    assert.ok(await element.isDisplayed());
});